tekton/ci/jobs/tekton-golang-coverage.yaml

```diff
@@ -161,85 +152,77 @@ spec:
-        - name: githubTokenFile
-          value: $(params.githubTokenFile)
-      workspaces:
+      taskSpec:
+        steps:
+          - name: clone
+            ref:
+              name: git-batch-merge
+            params:
+              - name: url
+                value: https://github.com/$(tasks.split-full-repo-name.results.repoOwner)/$(tasks.split-full-repo-name.results.repoName).git
+              - name: mode
+                value: "merge"
+              - name: revision
+                value: $(params.pullRequestBaseRef)
+              - name: refspec
+                value: refs/heads/$(params.pullRequestBaseRef):refs/heads/$(params.pullRequestBaseRef)
+              - name: batchedRefs
+                value: "refs/pull/$(params.pullRequestNumber)/head"
+              - name: sourcePath
+                value: $(workspaces.source.path)
+          - name: coverage
+            ref:
+              name: go-coverage
+            params:
+              - name: pullRequestNumber
+                value: $(params.pullRequestNumber)
+              - name: gitRevision
+                value: $(params.gitRevision)
+              - name: buildUUID
+                value: $(params.buildUUID)
+              - name: repoName
+                value: $(tasks.split-full-repo-name.results.repoName)
+              - name: repoOwner
+                value: $(tasks.split-full-repo-name.results.repoOwner)
+              - name: jobType
+                value: $(params.jobType)
+              - name: jobName
+                value: $(params.jobName)
+              - name: postSubmitGcsBucket
+                value: $(params.postSubmitGcsBucket)
+              - name: postSubmitJobName
+                value: $(params.postSubmitJobName)
+              - name: covTarget
+                value: $(params.covTarget)
+              - name: profileName
+                value: $(params.profileName)
+              - name: covThresholdPercentage
+                value: $(params.covThresholdPercentage)
+              - name: githubTokenFile
+                value: $(workspaces.github-token.path)/$(params.githubTokenFile)
+              - name: sourcePath
+                value: $(workspaces.source.path)
+          - name: upload
+            ref:
+              name: gcs-upload
+            params:
+              - name: path
+                value: $(workspaces.source.path)/artifacts
+              - name: location
+                value: gs://$(params.uploadGcsBucket)/pr-logs/pull/$(tasks.split-full-repo-name.results.repoOwner)_$(tasks.split-full-repo-name.results.repoName)/$(params.pullRequestNumber)/$(params.jobName)/$(params.buildUUID)/artifacts
+              - name: credentialsPath
+                value: $(workspaces.credentials.path)
+        workspaces:
         - name: source
-          workspace: source
+          description: Location where the git repo will be installed.
+          mountPath: /go/src/github.com/tektoncd/$(params.repoName)
```

Comment:
Do we still need that 😛 ? (aka the `GOPATH`)

Reply:
I'm not sure 🤔 . 

I wanted to limit the changes to just replace Task with StepAction for dogfooding purposes. I could check this in a follow-up PR and fix there if this is no longer needed. For now, I wanted to make sure that we can execute this as a StepAction on the Dogfooding cluster.